pwnlib/encoders/encoder.py

```diff
@@ -97,7 +97,7 @@ def encode(raw_bytes, avoid=None, expr=None, force=0, pcreg=''):
     elif expr:
         avoid_errmsg = repr(expr)
     else:
-        avoid_errmsg = ''.join(avoid)
+        avoid_errmsg = ''.join(repr(avoid))
```

Comment:
`''.join` is redundant on `repr(avoid)`, since it is already a string; you probably meant
```suggestion
        avoid_errmsg = repr(bytes(avoid))
```
which would make sure that it prints representations of the bytes, and not e.g. a list (set?) of numbers; I don't see whether `avoid` can be a set of byte strings, maybe it could be, but should be normalized to a `bytearray` anyway in the ideal case